### ISSUE TITLE
Evaluation path too deep

### ISSUE THREAD

**Issue by rapw3k** [RDFLib/pySHACL#70]
Hi,
I am trying to use the pyshacl to validate some data graphs against our SHACL specification. I like the options that pyshacl provides to include for example inferencing, allow imports or linking to the ontology. 

However I cannot make even simple validation work.

pyshacl -s <path-to-shape-graph> -m -i rdfs -a -j -f human <path-to-data-graph>

The shape is here: https://raw.githubusercontent.com/rapw3k/DEMETER/master/models/SHACL/demeterAgriProfile-SHACL.ttl 
The example data graph is here: https://box.psnc.pl/f/c95eb51962/?raw=1 

I validated the shape in correct using https://www.itb.ec.europa.eu/shacl/shacl/upload 

thanks

**Comment by ashleysommer** [comment]
Hi @rapw3k 
Thanks for the report. 
It looks like a change in the master shacl.ttl file published by the W3C has broken PySHACL's "MetaSHACL" feature. That is causing the "Evaluation Path Too Deep" error you are seeing. That bug will be fixed in the next version of PySHACL.

However, you can get around this by simply removing the `-m` flag from your run command. That will turn off the MetaSHACL feature, and allow your validation work.

Except, after trying that, I am now seeing an error in your SHACL file.
>pyshacl.errors.ShapeLoadError: A shape defined as a PropertyShape must be the subject of a 'sh:path' predicate.

In your Shapes file, the `PropertyShape` at `<https://astrea.linkeddata.es/shapes#7c708841b8b6a556648a36e2070312bc>` does not have a  `sh:path` property that is causing pySHACL to throw a ShapeLoad error.

**Comment by ashleysommer** [comment]
fixed by https://github.com/RDFLib/pySHACL/commit/4fcf5d9c9de9ec0b4f25c2e0de9f7620ae6ca820
Released as v0.14.3

**Comment by rapw3k** [comment]
Hi @ashleysommer 
Thanks a lot for the hints.

**Issue 1**
I checked those erros in the shapefile of property shapes without sh:path. These were generated automatically for owl:disjointwith statements, which translate it into the statements below (https://astrea.linkeddata.es/documentation.html). 
```
?shapeUrl a sh:NodeShape ;
   sh:not ?propertyShape . 
?propertyShape a sh:PropertyShape ;
   sh:class ?disjointType .
```
I understand this is not correct, so I can just remove the statement of type and it works.
```
?shapeUrl a sh:NodeShape ;
   sh:not ?propertyShape . 
?propertyShape 
   sh:class ?disjointType .
```
An issue, however, with the command line is that I didnt get information of the error, so I didnt know what was happening. The error i got is runtime error (MacOS) as below:
```
RAPMAC-3:SHACL rap$ pyshacl -s /Users/rap/GitRepositories/GitHub/DEMETER/models/SHACL/demeterAgriProfile-SHACL.ttl -e /Users/rap/GitRepositories/GitHub/DEMETER/models/cross-domain.ttl -i rdfs -a -j -f human /Users/rap/Downloads/pilot5.2-afc-observation-point-simplified.ttl
  File "/usr/local/lib/python3.9/site-packages/pyshacl/cli.py", line 169, in main
    is_conform, v_graph, v_text = validate(args.data, **validator_kwargs)
  File "/usr/local/lib/python3.9/site-packages/pyshacl/validate.py", line 390, in validate
    conforms, report_graph, report_text = validator.run()
  File "/usr/local/lib/python3.9/site-packages/pyshacl/validate.py", line 223, in run
    shapes = self.shacl_graph.shapes  # This property getter triggers shapes harvest.
  File "/usr/local/lib/python3.9/site-packages/pyshacl/shapes_graph.py", line 164, in shapes
    self._build_node_shape_cache()
  File "/usr/local/lib/python3.9/site-packages/pyshacl/shapes_graph.py", line 208, in _build_node_shape_cache
    raise ShapeLoadError(


Validator encountered a Runtime Error. Please report this to the PySHACL issue tracker.
```
**issue 2**
I fixed the shape, accessible in the same location above, and tried adding the target ontology into the mix with the -e option.
The target ontology is available here: https://raw.githubusercontent.com/rapw3k/DEMETER/master/models/cross-domain.ttl 
Now, I get a validation error:
```
RAPMAC-3:SHACL rap$ pyshacl -s /Users/rap/GitRepositories/GitHub/DEMETER/models/SHACL/demeterAgriProfile-SHACL.ttl -e /Users/rap/GitRepositories/GitHub/DEMETER/models/cross-domain.ttl -i rdfs -a -j -f human /Users/rap/Downloads/pilot5.2-afc-observation-point-simplified.ttl
Validation Report
Conforms: False
Results (1):
Constraint Violation in DatatypeConstraintComponent (http://www.w3.org/ns/shacl#DatatypeConstraintComponent):
	Severity: sh:Violation
	Source Shape: <https://astrea.linkeddata.es/shapes#0f6502053608fc059a721676b6e67115>
	Focus Node: <http://www.w3id.org/afarcloud/pCoord?lat=45.75&amp;long=4.85>
	Value Node: Literal("POINT(45.75 4.85)" = None, datatype=ns0:wktLiteral)
	Result Path: geo:hasSerialization
	Message: Value is not Literal with datatype rdfs:Literal
```
This is because the data graph says that:

```
@prefix ns0: <http://www.opengis.net/ont/geosparql#> .
ns0:asWKT "POINT(45.75 4.85)"^^ns0:wktLiteral
```
The ontology says that asWKT has range wktLiteral and it is a subproperty of hasSerialization, which has range rdfs:Literal. Additionally, the ontology defines wktLiteral as a Datatype (and according to spec Each instance of rdfs:Datatype is a subclass of rdfs:Literal). So, I dont really understand why the validation error?

```
@prefix geo: <http://www.opengis.net/ont/geosparql#> .
geo:asWKT rdf:type owl:DatatypeProperty ;
          rdfs:subPropertyOf geo:hasSerialization ;
          rdfs:domain geo:Geometry ;
          rdfs:range geo:wktLiteral ;
...
geo:hasSerialization rdf:type owl:DatatypeProperty ;
                     rdfs:domain geo:Geometry ;
                     rdfs:range rdfs:Literal ;
...
geo:wktLiteral rdf:type rdfs:Datatype ;
               rdfs:comment "A Well-known Text serialization of a geometry object."@en ;


```
